### DISCUSSION TITLE
How to get the last item of a list?

### DISCUSSION THREAD

**Discussion by YoRHa-A5** [pocketbase/pocketbase#1530] [Q&A] [answered]
Hello,

There is `.getFirstListItem()` for getting the first item of a collection that matches a specific filter.
How can one get the last item of a collection? Currently I use

```js
const list = await pb.collection('collection').getFullList(undefined, {
	sort: '-created'
});
lastItem = list.at(-1);
```

Is there a better way? Since I'm not sure the `getFullList()` will run into problems/take too long once there are too many items in a collection?

Thanks 🙂

**Comment by ganigeorgiev** [comment, marked as answer]
`getFullList()` will query all items.

If you want only a single item, you can use `getList(1, 1)` (internally `getFullList()` calls `getList()` until all items are fetched):

```js
const result = await pb.collection('example').getList(1, 1, { sort: '-created' });
const lastItem = result.items[0];
```

Alternatively, you can also try with `pb.collection('example').getFirstListItem(null, { sort: '-created' })` (or instead of `null` you can also pass a dummy filter like '1=1').
